### DISCUSSION TITLE
Deprecate the use of deep lists when composing styles in favor of the Spread syntax

### DISCUSSION THREAD

**Discussion by UpperCod** [atomicojs/atomico#57] [Ideas]
In Atomico 1.31.0 the compatibility with deep list composition exploration of style was announced, this in order to have an aesthetic similar to the use of children.

This was an early feature and I leave the space to discuss the disconuity of this feature.

# 1.31.0 

```js
const style1 = css`:host{}`;
const style2 = [style1, css`:host{}` ];

function component(){
    return <host></host>;
}

component.styles = [
    style2,
    css`:host{...}`
]
```

# before 1.31.0 

```js
const style1 = css`:host{}`;
const style2 = [style1, css`:host{}` ];

function component(){
    return <host></host>;
}

component.styles = [
   ...style2, // 👈 Spread syntax 
   css`:host{...}`
]
```

## cost without compression

`151bytes` aprox 

### before
```js
//fragment 1
return [super.styles||[],styles||[]]
//fragment 2
styles = new Set(...styles);
```
### 1.31.0
```js
//fragment 1
return [super.styles,styles]
//fragment 2
styles = styles.reduce(function concat(styles, style) {
      isArray(style)
          ? style.reduce(concat, styles)
          : style && styles.push(style);
      return styles;
}, []);
```

**Comment by Explosion-Scratch** [comment]
Why are you doing the weird reduce stuff, just do [`styles.flat(1000)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat)

**Reply by UpperCod** [reply]
The reality is that it did not know the first parameter for `[].flat` and in my mind reduce responded to the analysis of empty and lists before inserting ... Now comparing there are [differences in performance](https://perf.link/#eyJpZCI6Im5kbGh4YXBmd3MyIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoMTAwMCkua2V5cygpLC4uLkFycmF5KDEwMDApLmtleXMoKV07IiwidGVzdHMiOlt7Im5hbWUiOiJGaW5kIGl0ZW0gMTAwIiwiY29kZSI6ImRhdGEuZmxhdCgxMDAwKS5maWx0ZXIoKHZhbHVlKT0%2BdmFsdWUpIiwicnVucyI6WzMwMDAsNDAwMCwzMzMzLDQwMDAsMzY2NiwyNjY2LDMwMDAsMzAwMCwzMDAwLDMzMzMsMzY2NiwzMDAwLDQwMDAsNDAwMCwyNjY2LDMwMDAsMzAwMCwzMDAwLDI2NjYsMzAwMCwzNjY2LDMwMDAsMzMzMywzMzMzLDMzMzMsMzAwMCwzMDAwLDM2NjYsMjY2NiwyNjY2LDMzMzMsMzAwMCw0MDAwLDMwMDAsNDAwMCw0MDAwLDM2NjYsMzY2NiwzMzMzLDM2NjYsMzAwMCwzMDAwLDMwMDAsMzY2NiwyNjY2LDMwMDAsMzAwMCwyNjY2LDM2NjYsMzY2NiwzNjY2LDM2NjYsMzAwMCwyNjY2LDMwMDAsMjY2NiwzMDAwLDMzMzMsMjY2NiwzMDAwLDMwMDAsMzY2NiwzNjY2LDMwMDAsMjY2NiwzMDAwLDMwMDAsMzMzMywzNjY2LDMwMDAsMzAwMCwzMDAwLDMzMzMsMzY2NiwzMDAwLDMwMDAsMzMzMywzNjY2LDQwMDAsNDAwMCwzMzMzLDMwMDAsMzAwMCwzMDAwLDMwMDAsMzAwMCw0MDAwLDMwMDAsMjY2NiwzMzMzLDMzMzMsMzY2NiwzNjY2LDMwMDAsMzAwMCwyNjY2LDMwMDAsMjY2NiwzMzMzLDMwMDBdLCJvcHMiOjMyMjl9LHsibmFtZSI6IkZpbmQgaXRlbSAyMDAiLCJjb2RlIjoiZGF0YS5yZWR1Y2UoZnVuY3Rpb24gY29uY2F0KHN0eWxlcywgc3R5bGUpIHtcbiAgICBBcnJheS5pc0FycmF5KHN0eWxlKVxuICAgICAgICA%2FIHN0eWxlLnJlZHVjZShjb25jYXQsIHN0eWxlcylcbiAgICAgICAgOiBzdHlsZSAmJiBzdHlsZXMucHVzaChzdHlsZSk7XG4gICAgcmV0dXJuIHN0eWxlcztcbn0sIFtdKTsiLCJydW5zIjpbMTc2NjYsMjAzMzMsMTUwMDAsMjAzMzMsMTc2NjYsMTgzMzMsMTQwMDAsMjIwMDAsMjMzMzMsMTYwMDAsMTczMzMsMjM2NjYsMTkwMDAsMjAzMzMsMTc2NjYsMTMzMzMsMjA2NjYsMTIwMDAsOTAwMCwxMzY2NiwxNzY2NiwxNTAwMCwxNzMzMywxNTY2NiwxNzMzMywyMTMzMywxMjMzMywxNzY2NiwyMjY2NiwyMjAwMCwxNDY2NiwxNDAwMCwyMTY2NiwxNDMzMywyMTMzMywyMDMzMywxODMzMywxNjMzMywxNjMzMywxNzY2NiwyMzY2NiwxMjAwMCwxMjY2NiwxNzMzMywyMDMzMywxNDMzMywyMTMzMywxNDAwMCwyMTY2NiwxNzY2NiwxODY2NiwxODMzMywxMzY2NiwyMDAwMCwyMjMzMywyMTAwMCwyMTY2NiwxNTMzMywyMDAwMCwxMzY2NiwxNDMzMywxODY2NiwxNzY2NiwyMzAwMCwxNzY2NiwxMzAwMCwxMzMzMywxNjAwMCwxNzY2NiwyNDMzMywxNjAwMCwyMzY2NiwxNjAwMCwxOTY2NiwxMjMzMywxNDMzMywxNzMzMywxOTY2NiwyNDY2NiwyMDAwMCwxNzAwMCwxNjMzMywyMTAwMCwxMjMzMywxMzMzMywyMTAwMCwyMDY2NiwxMjY2NiwyMDAwMCwxNTY2NiwxNzAwMCwxODMzMywyMjY2NiwyMzAwMCwxMDMzMywyMjMzMywxMDAwMCwyMTAwMCwxMjMzMywyMjY2Nl0sIm9wcyI6MTc3Mzl9XSwidXBkYXRlZCI6IjIwMjEtMDktMTdUMjI6NTY6NTAuMjQ4WiJ9):

3,229 ops / s flat
17,739 ops / s reduces

.... Performance or Simplicity? that's the dilemma

By the way I do not rule out the solution with `.flat`, it would be good to use it eliminating the empty condition

**Reply by Explosion-Scratch** [reply]
Still, you're not going to be flattening huge lists, right? Theoretically these are going to be small, and atomico advertises being small, so using `.flat` supports that ideology 😐 Ultimately it's a small issue and up to you as the developer, just my thoughts! 😃
